// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: model/model.proto

package model

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on ClusterModel with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *ClusterModel) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ClusterModel with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in ClusterModelMultiError, or
// nil if none found.
func (m *ClusterModel) ValidateAll() error {
	return m.validate(true)
}

func (m *ClusterModel) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Name

	// no validation rules for KubeConfig

	// no validation rules for ApiServerUrl

	// no validation rules for CreatedAt

	// no validation rules for UpdatedAt

	// no validation rules for DeletedAt

	if len(errors) > 0 {
		return ClusterModelMultiError(errors)
	}

	return nil
}

// ClusterModelMultiError is an error wrapping multiple validation errors
// returned by ClusterModel.ValidateAll() if the designated constraints aren't met.
type ClusterModelMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ClusterModelMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ClusterModelMultiError) AllErrors() []error { return m }

// ClusterModelValidationError is the validation error returned by
// ClusterModel.Validate if the designated constraints aren't met.
type ClusterModelValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ClusterModelValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ClusterModelValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ClusterModelValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ClusterModelValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ClusterModelValidationError) ErrorName() string { return "ClusterModelValidationError" }

// Error satisfies the builtin error interface
func (e ClusterModelValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sClusterModel.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ClusterModelValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ClusterModelValidationError{}

// Validate checks the field values on CardModel with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *CardModel) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CardModel with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in CardModelMultiError, or nil
// if none found.
func (m *CardModel) ValidateAll() error {
	return m.validate(true)
}

func (m *CardModel) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Type

	// no validation rules for Namespace

	// no validation rules for Name

	// no validation rules for ClusterId

	// no validation rules for CreatedAt

	// no validation rules for UpdatedAt

	// no validation rules for DeletedAt

	if len(errors) > 0 {
		return CardModelMultiError(errors)
	}

	return nil
}

// CardModelMultiError is an error wrapping multiple validation errors returned
// by CardModel.ValidateAll() if the designated constraints aren't met.
type CardModelMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CardModelMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CardModelMultiError) AllErrors() []error { return m }

// CardModelValidationError is the validation error returned by
// CardModel.Validate if the designated constraints aren't met.
type CardModelValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CardModelValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CardModelValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CardModelValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CardModelValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CardModelValidationError) ErrorName() string { return "CardModelValidationError" }

// Error satisfies the builtin error interface
func (e CardModelValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCardModel.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CardModelValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CardModelValidationError{}

// Validate checks the field values on FileModel with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *FileModel) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on FileModel with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in FileModelMultiError, or nil
// if none found.
func (m *FileModel) ValidateAll() error {
	return m.validate(true)
}

func (m *FileModel) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Path

	// no validation rules for Size

	// no validation rules for Username

	// no validation rules for Namespace

	// no validation rules for Pod

	// no validation rules for Container

	// no validation rules for ContainerPath

	// no validation rules for CreatedAt

	// no validation rules for UpdatedAt

	// no validation rules for DeletedAt

	// no validation rules for IsDeleted

	if len(errors) > 0 {
		return FileModelMultiError(errors)
	}

	return nil
}

// FileModelMultiError is an error wrapping multiple validation errors returned
// by FileModel.ValidateAll() if the designated constraints aren't met.
type FileModelMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m FileModelMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m FileModelMultiError) AllErrors() []error { return m }

// FileModelValidationError is the validation error returned by
// FileModel.Validate if the designated constraints aren't met.
type FileModelValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e FileModelValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e FileModelValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e FileModelValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e FileModelValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e FileModelValidationError) ErrorName() string { return "FileModelValidationError" }

// Error satisfies the builtin error interface
func (e FileModelValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sFileModel.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = FileModelValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = FileModelValidationError{}
